src/ejabberd_tls.erl

```diff
@@ -135,11 +143,16 @@ has_peer_cert(Opts) ->
             true
     end.
 
-get_peer_cert(fast_tls, Socket) ->
-    case {fast_tls:get_verify_result(Socket), fast_tls:get_peer_certificate(Socket)} of
-        {0, {ok, Cert}} -> {ok, Cert};
-        {Error, {ok, Cert}} -> {bad_cert, fast_tls:get_cert_verify_string(Error, Cert)};
-        {_, error} -> no_peer_cert
+%% 18 is OpenSSL's and fast_tls's error code for self-signed certs
+maybe_self_signed(18 = Error, Cert, SSLOpts) ->
```

Comment:
Without reading through the implementation of this & `get_peer_certificate` I didn't understand what this function should be doing - could the name be a bit more descriptive?

Reply:
It probably could :) I'm open to suggestions on a better name :)

Reply:
What I think it's doing, `maybe_allow_selfsigned`?

Reply:
Sounds good I pushed the change.